src/dataframe/insertion.jl

```diff
@@ -762,16 +766,29 @@ function _dfr_row_inserter!(df::DataFrame, loc::Integer, dfr::DataFrameRow,
     return _row_inserter!(df, loc, dfr, mode, cols, promote, nrows)
 end
 
+_abstractrow2namedtuple(row::Tables.AbstractRow) =
```

Comment:
I assume that these methods are not used for super wide tables (like 10 000 columns, so this fallback should be good enough).

If we feel that this is not the case we can have some more fancy code for this, but I wanted to start with something simple.

Reply:
I agree, I think starting simple is a good way to go 👍

Reply:
It seems like we support `AbstractDict` as well, so would it be worth doing a quick check on the `AbstractRow` length and then we support both out of the box? Something like:

```julia
Base.push!(df::DataFrame, row::Tables.AbstractRow) = length(Tables.columnnames(row)) > 100 ? _abstractrow2dict(row) : _abstractrow2namedtuple(row)
```

Reply:
Ah, that's a good point. Carry on then.

Reply:
Well, actually, I'll mention that in Tables.jl we took the dependency of https://github.com/JuliaCollections/OrderedCollections.jl, which is a much smaller, lighter dependency than DataStructures.jl. So I believe that should already be available in DataFrames.jl?

Reply:
Good point - I will check. Probably you are right 😄.

Reply:
Actually using `OrderedDict` turned out to be faster than `NamedTuple` overall for some reason (even after compilation).
I also made some small adjustment to better support `OrderedDict` in general, since we support it.